src/class-wpml-gutenberg-integration-factory.php

```diff
@@ -31,18 +31,24 @@ public function create() {
 			)
 		);
 
-		$reusable_blocks_translation = new WPML\PB\Gutenberg\Reusable_Blocks_Translation( $sitepress );
+		$reusable_blocks             = new WPML\PB\Gutenberg\ReusableBlocks\Blocks();
+		$reusable_blocks_translation = new WPML\PB\Gutenberg\ReusableBlocks\Translation( $sitepress );
 
 		$integrations->add(
-			new WPML\PB\Gutenberg\Reusable_Blocks_Integration( $reusable_blocks_translation )
+			new WPML\PB\Gutenberg\ReusableBlocks\Integration( $reusable_blocks_translation )
 		);
 
 		if ( is_admin() ) {
 			$integrations->add(
-				new WPML\PB\Gutenberg\Reusable_Blocks_Admin_Integration(
-					new WPML\PB\Gutenberg\Reusable_Blocks_Batch_Handler(
-						new WPML\PB\Gutenberg\Reusable_Blocks(),
+				new WPML\PB\Gutenberg\ReusableBlocks\AdminIntegration(
+					new WPML\PB\Gutenberg\ReusableBlocks\ManageBatch(
+						$reusable_blocks,
 						$reusable_blocks_translation
+					),
+					new WPML\PB\Gutenberg\ReusableBlocks\ManageBasket(
+						$reusable_blocks,
+						$reusable_blocks_translation,
+						\WPML\Container\make( '\WPML_Translation_Basket' )
```

Comment:
@strategio Since you are using DIC here, I think we should use it correctly.

This whole code can be replaced with:
```
$integrations->add( \WPML\Container\make( 'WPML\PB\Gutenberg\ReusableBlocks\Integration' );
if ( is_admin() ) {
   $integrations->add( \WPML\Container\make( 'WPML\PB\Gutenberg\ReusableBlocks\AdminIntegration' );
}
```
BTW, What happens if TM is not active and `WPML_Translation_Basket` doesn't exist?

Reply:
@brucepearson, I implemented your suggestion in https://github.com/OnTheGoSystems/wpml-page-builders-gutenberg/pull/38.